File: Capítulo_5/Deque/Removendo elementos na frente do deque/removeFront.js
// removeFront(): esse método remove o primeiro elemento do deque.
class Deque {
    constructor() {
        this.count = 0;
        this.lowestCount = 0;
        this.items = {};
    }
    
    removeFront() {
        if (this.isEmpty()){
            return undefined;
        }
        const result = this.items[this.lowestCount];
        delete this.items[this.lowestCount];
        this.lowestCount++;
        return result;
    }
}

File: Capítulo_5/Deque/Removendo elementos na frente do deque/Código em Python/removeFront.py
# removeFront(): esse método remove o primeiro elemento do deque.
class Deque:
    def __init__(self):
        self.count = 0
        self.lowest_count = 0
        self.items = {}
    
    def removeFront(self):
        if (self.is_empty()):
            return None
        result = self.items[self.lowest_count]
        del self.items[self.lowest_count]
        self.lowest_count += 1
        return result
    